analysis/analysis/RegisterAllocationAnalysis.java

```
@@ -0,0 +1,83 @@
+package analysis;
+
+import petter.cfg.AbstractPropagatingVisitor;
+import petter.cfg.CompilationUnit;
+import petter.cfg.State;
+import petter.cfg.edges.ProcedureCall;
+import petter.cfg.expression.Variable;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RegisterAllocationAnalysis extends AbstractPropagatingVisitor<Map<Variable, Integer>> {
+    CompilationUnit cu;
+    TrueLivenessAnalysis la;
+
+    public RegisterAllocationAnalysis(CompilationUnit cu, TrueLivenessAnalysis la) {
+        super(true);
+        this.cu = cu;
+        this.la = la;
+    }
+
+    public Map<Variable, Integer> visit(ProcedureCall procedureCall, Map<Variable, Integer> objectObjectMap) {
+        return null;
```

Did you test this when the C code contains a function call?
My guess would be that the analysis stops on the first call (though I’m not completely sure).

---

Did we handle procedure calls, when it comes to register analysis. I don't think we did, this is why I didn't do it, but I can just propagate the prev value, instead of passing null

---

Good point, I guess implementing calling conventions would be done in another compilation step so passing sounds good.